### PR TITLE
add tolerations to aad pod identity

### PR DESCRIPTION
**Pull request by sonasingh46**
Signed-off-by: Ashutosh Kumar <sonasingh46@gmail.com>

### What this PR does / why we need it

Add tolerations to aad-pod-identity component YAML.

CAPZ pods has control plane tolerations and this means that capz pod can get scheduled to control plane nodes. The aad-pod-identity does not have tolerations which can cause authentication issue if the capz pod gets scheduled to a ctonrol plane node. 
This PR adds control plane tolerations to aad pod identity so that the their pods can be scheduled to control plane nodes too.

### Which issue(s) this PR fixes

Fixes #

### Release note

```release-note

```

### CODE REVIEW

**Comment by rajathagasthya** [comment]
PR needs an issue linked (create one if it doesn't exist) and please describe how the PR is tested.

File: hack/providers-sync-tools/cluster-api-provider-azure/overlay/add-tolerations.yaml [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+spec:
+  template:
+    spec:
+      #@overlay/match missing_ok=True
+      tolerations:
+        - effect: NoSchedule
+          key: node-role.kubernetes.io/master
+        - effect: NoSchedule
+          key: node-role.kubernetes.io/control-plane
+
```
Comment (wjun) [inline hack/providers-sync-tools/cluster-api-provider-azure/overlay/add-tolerations.yaml:15]: remove this line

**Comment by wjun** [comment]
Please use a non-forked repo so the pr can be tested by capz/capa cc pipelines' integration tests.

**Comment by sonasingh46** [comment]
@wjun -- Closing this PR in favour of ( for non fork branch PR )
https://github.com/vmware-tanzu/tanzu-framework/pull/3947

File: providers/infrastructure-azure/v1.5.3/infrastructure-components.yaml [hunk 2]
```diff
@@ -8638,6 +8638,11 @@ spec:
           path: /etc/default/kubelet
           type: FileOrCreate
         name: kubelet-config
+      tolerations:
```
[the remark this answers is not in the document]
Comment (sonasingh46) [reply providers/infrastructure-azure/v1.5.3/infrastructure-components.yaml:8641]: I do not know about this. Please check with cli team or relevant people who knows about how the dependency is baked in.

But after I tested the changes - I have the appropriate results.

**Pull request closed without merging**